### PR TITLE
Refactor UI components to function components

### PR DESCRIPTION
**Pull request by whiteHatCip**
# Overview
This MR aims at providing a new implementation of the UI components that makes use of function components, moving away from the class components that were already in place.

# Test Plan

### CODE REVIEW

File: src/components/Overlay.tsx [hunk 1]
```diff
@@ -23,41 +23,28 @@ export interface OverlayProps {
   visible: boolean;
 }
 
-interface State {
-  fadeAnim: Animated.Value;
-  overlayStyle: StyleProp<ViewStyle>;
-}
+const Overlay: React.FC<React.PropsWithChildren<OverlayProps>> = ({ visible, children }) => {
+  const [fadeAnim] = React.useState(new Animated.Value(0));
+  const [overlayStyle, setOverlayStyle] = React.useState<StyleProp<ViewStyle>>(styles.emptyOverlay);
```
Comment (MateusAndrade) [inline src/components/Overlay.tsx:28]: What about importing the `useState` separately from React?

Comment (whiteHatCip) [reply src/components/Overlay.tsx:28]: > What about importing the `useState` separately from React? 

We can do that, of course. But I don't see that much difference imho. Will make the change when I'm back from holidays tho😉

**Review by MateusAndrade — APPROVED** [review]

**Comment by whiteHatCip** [comment]
The docs page should remove the "Refactor to function components" element from the todo list at this link:
https://react-native-share.github.io/react-native-share/docs/ui-components

**Comment by MateusAndrade** [comment]
> The docs page should remove the "Refactor to function components" element from the todo list at this link: https://react-native-share.github.io/react-native-share/docs/ui-components

Done: https://github.com/react-native-share/react-native-share/commit/fd94ea54f74dd17d054b760256023ece351f9296! 🚀

**Pull request merged**